#!/usr/bin/env python
PACKAGE = "perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#int
gen.add("minClusterSize_",    int_t,    0, "An Integer parameter", 80,  1, 1000)

#double
gen.add("clusterRadius_",    double_t,    0, "An double parameter", 0.8,  0.01, 5)


#bool
# gen.add("bool_outlier_removal",   bool_t,   0, "A Boolean parameter",  True)


#string
# gen.add("save_name",    str_t,    0, "A string parameter",  "intersection_info")

# size_enum = gen.enum([ gen.const("Low",      int_t, 0, "Low is 0"),
#                        gen.const("Medium",     int_t, 1, "Medium is 1"),
#                        gen.const("High",      int_t, 2, "Hight is 2")],
#                      "Select from the list")

# gen.add("size", int_t, 0, "Select from the list", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "perception", "laneDetection_"))
